src/reflect/type.go

```diff
@@ -498,7 +498,13 @@ func pointerTo(t *rawType) *rawType {
 
 func (t *rawType) String() string {
 	if t.isNamed() {
-		return t.PkgPath() + "." + t.Name()
+		path := t.PkgPath()
+		slash := len(path) - 1
+		for slash >= 0 && path[slash] != '/' {
+			slash--
+		}
+		shortened := path[slash+1:]
+		return shortened + "." + t.Name()
```

Comment:
You're right.  I'll fix this.

Reply:
Since PkgPath and packge name might not be the same, we'll need to store that field separately for the `namedType` struct.  We might be able to do something sneaky here to save space.

Reply:
Sample patch (compiler changes not included):
```
@@ -1026,10 +1020,21 @@ func readStringZ(data unsafe.Pointer) string {
        }))
 }

+func (t *rawType) name() string {
+       ntype := (*namedType)(unsafe.Pointer(t))
+       return readStringZ(unsafe.Pointer(&ntype.name[0]))
+}
+
 func (t *rawType) Name() string {
        if t.isNamed() {
-               ntype := (*namedType)(unsafe.Pointer(t))
-               return readStringZ(unsafe.Pointer(&ntype.name[0]))
+               pkgDotName := t.name()
+               // strip the package and the `.` that appear before the type name
+               for i := 0; i < len(pkgDotName); i++ {
+                       if pkgDotName[i] == '.' {
+                               return pkgDotName[i+1:]
+                       }
+               }
+               panic("corrupt name field for named type")
```